modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalRequest.java

```diff
@@ -111,12 +118,38 @@ public void indicesOptions(IndicesOptions indicesOptions) {
         this.indicesOptions = Objects.requireNonNull(indicesOptions, "indicesOptions must not be null");
     }
 
+    /**
+     * The search type to execute, defaults to {@link SearchType#DEFAULT}.
+     */
+    public void searchType(SearchType searchType) {
+        this.searchType = Objects.requireNonNull(searchType, "searchType must not be null");
+    }
+
+    /**
+     * The tye of search to execute.
+     */
+    public SearchType searchType() {
+        return searchType;
+    }
+
+    /**
+     * The a string representation search type to execute, defaults to {@link SearchType#DEFAULT}. Can be one of
+     * "dfs_query_then_fetch"/"dfsQueryThenFetch", "dfs_query_and_fetch"/"dfsQueryAndFetch", "query_then_fetch"/"queryThenFetch", and
+     * "query_and_fetch"/"queryAndFetch".
+     */
```

Comment:
Is this really needed ? It should be enough to use the `SearchType` enum ? If not, then the comment should be changed since the types are restricted to `dfs_query_then_fetch` and `query_then_fetch`.